components/TextFields/src/ContainedInputView/private/MDCContainedInputViewStyleBase.m

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCContainedInputViewStyleBase.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "MDCContainedInputView.h"
+#import "MDCContainedInputViewVerticalPositioningGuideBase.h"
+
+static const CGFloat kBaseFloatingLabelScaleFactor = 0.75;
+
+@implementation MDCContainedInputViewStyleBase
+
+- (UIFont *)floatingFontWithFont:(UIFont *)font {
+  CGFloat scaleFactor = kBaseFloatingLabelScaleFactor;
+  CGFloat floatingFontSize = font.pointSize * scaleFactor;
+  return [font fontWithSize:floatingFontSize];
+}
+
+- (void)applyStyleToContainedInputView:(id<MDCContainedInputView>)inputView {
+}
+
+- (void)removeStyleFrom:(id<MDCContainedInputView>)containedInputView {
+}
+
+- (id<MDCContainerStyleVerticalPositioningReference>)positioningReference {
+  return [[MDCContainedInputViewVerticalPositioningGuideBase alloc] init];
```

Comment:
It seems a little odd to create a new one every time we are asked for one. Im also trying to understand the value of the MDCContainedInputViewVerticalPositioningGuideBase object having readonly properties.

Reply:
There is a good amount of stuff related to the positioning reference that's yet to be brought over. Eventually, this object is going to have a more complicated initializer that will accept things like the floating label height, for example. The height of the floating label is liable to change if someone does something like set the font on the textfield. I think passing in these values as they exist at a given point in time to get a readonly object whose values we know are right ends up being preferable to remembering to mutate an object in every setter that might impact it.